utils/cli/core/core.py

```diff
@@ -198,6 +198,54 @@ def tasks_upload(self, task_id, fileformat, filename, **kwargs):
             "with annotation file {} finished".format(filename)
         log.info(logger_string)
 
+    def tasks_export(self, task_id, filename, export_verification_period=3, **kwargs):
+        """ Export and download a whole task """
+        url = self.api.tasks_id(task_id)
+        export_url = url + '?action=export'
+        response = self.session.get(export_url)
```

Comment:
Looks like exactly the lines are in the beginning of while loop. Move sleep at the end of while and you can remove these 2 lines.